app/src/app/utils/helpers.ts

```diff
@@ -119,66 +116,19 @@ export const getFeaturesIntersectingCounties = (
   });
 
   if (!countyFeatures) return;
+  const fips = new Set<string>(countyFeatures.map(c => c.properties.STATEFP + c.properties.COUNTYFP));
 
-  const featureBbox = getBoundingBoxFromFeatures(countyFeatures);
-
-  if (!featureBbox) return;
-
-  const sw = map.project(featureBbox[0]);
-  const ne = map.project(featureBbox[1]);
-  const features = map.queryRenderedFeatures([sw, ne], {
+  const features = map.queryRenderedFeatures(undefined, {
     layers,
   });
 
-  let countyPoly;
-  try {
-    // @ts-ignore: Property 'coordinates' does not exist on type 'Geometry'.
-    countyPoly = polygon(countyFeatures[0].geometry.coordinates);
-  } catch {
-    // @ts-ignore: Property 'coordinates' does not exist on type 'Geometry'.
-    countyPoly = multiPolygon(countyFeatures[0].geometry.coordinates);
-  }
-
   return features.filter(p => {
-    const point = pointOnFeature(p);
-    return booleanWithin(point, countyPoly);
+    if (!(p?.id)) return false;
+    const myFips = p.id.toString().match(/\d{5}/);
+    return myFips && fips.has(myFips[0]);
```

Comment:
Q: This logic looks good and works in the app but feels a little laggy. `Set.has` should be a constant time operation which is why it's sort of weird that it would be the cause of the lag. We could try using [`Set.prototype.intersection()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set/intersection) like:
```javascript
return fips.intersection(
    new Set(features.filter(p => !(p?.id) &&  p.id.toString().match(/\d{5}/)))
)
```
... but I'm not sure if that's the issue or...

Q: What does querying `undefined` do? We only need to query the paintable layers so perhaps passing those IDs in will improve perf.

Reply:
`undefined` is the screen bounding box, undefined meaning the full viewport
I thought that { layers } in that call is limiting it to one layer, as we also have `layers: string[] = [BLOCK_HOVER_LAYER_ID]` in the function args

Reply:
> `undefined` is the screen bounding box

Ah, got it ty